Program runs without giving any error.

However, here is Valgrind output:

valgrind --leak-check=yes ./q6
==5800== Memcheck, a memory error detector
==5800== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==5800== Using Valgrind-3.20.0 and LibVEX; rerun with -h for copyright info
==5800== Command: ./q6
==5800==
==5800== Invalid read of size 4
==5800==    at 0x400581: main (q6.c:9)
==5800==  Address 0x51ea044 is 4 bytes inside a block of size 400 free'd
==5800==    at 0x4C2F9CF: free (vg_replace_malloc.c:884)
==5800==    by 0x400578: main (q6.c:7)
==5800==  Block was alloc'd at
==5800==    at 0x4C2D0E5: malloc (vg_replace_malloc.c:393)
==5800==    by 0x400568: main (q6.c:5)
==5800==
0
==5800==
==5800== HEAP SUMMARY:
==5800==     in use at exit: 0 bytes in 0 blocks
==5800==   total heap usage: 2 allocs, 2 frees, 1,424 bytes allocated
==5800==
==5800== All heap blocks were freed -- no leaks are possible
==5800==
==5800== For lists of detected and suppressed errors, rerun with: -s
==5800== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)

It clearly shows that at line 9 program is accessing 4 bytes from a memory location which was freed.
